sdk/src/asset_handlers/bmff_io.rs

```diff
@@ -1547,6 +1556,268 @@ impl RemoteRefEmbed for BmffIO {
+
+    let mut boxes = Vec::new();
+
+    if let Some(ftyp_tokens) = bmff_map.get("/ftyp") {
+        boxes.push(bmff_tree[ftyp_tokens[0]].data.clone());
+    }
+
+    if let Some(moov_tokens) = bmff_map.get("/moov") {
+        boxes.push(bmff_tree[moov_tokens[0]].data.clone());
+    }
+
+    Ok(boxes)
+}
+
+#[allow(dead_code)]
+struct SampleSizeBox {
+    pub sample_size: u32,
+    pub sample_count: u32,
+    pub entries: Vec<u32>,
+}
+
+#[allow(dead_code)]
+impl SampleSizeBox {
+    pub fn read_box(reader: &mut dyn CAIRead) -> Result<SampleSizeBox> {
+        // read header
+        let header = BoxHeaderLite::read(reader)
+            .map_err(|_err| Error::InvalidAsset("Bad BMFF".to_string()))?;
+        if header.name != BoxType::StszBox {
+            return Err(Error::InvalidAsset("Bad BMFF".to_string()));
+        }
+
+        // read extended header
+        let (_version, _flags) = read_box_header_ext(reader)?; // box extensions
+
+        // get size & count of samples
+        let sample_size = reader.read_u32::<BigEndian>()?;
+        let sample_count = reader.read_u32::<BigEndian>()?;
+
+        // read and patch offsets
+        let mut entries: Vec<u32> = Vec::new();
+        for _e in 0..sample_count {
+            let entry_size = reader.read_u32::<BigEndian>()?;
+
+            entries.push(entry_size);
+        }
+
+        Ok(SampleSizeBox {
+            sample_size,
+            sample_count,
+            entries,
+        })
+    }
+}
+
+pub struct ChunkBox {
+    pub first_chunk: u32,
+    pub samples_per_chunk: u32,
+    pub sample_description_index: u32,
+}
+
+#[allow(dead_code)]
+struct SampleToChunkBox {
+    pub entry_count: u32,
+    pub entries: Vec<ChunkBox>,
+}
+
+impl SampleToChunkBox {
+    pub fn read_box(reader: &mut dyn CAIRead) -> Result<SampleToChunkBox> {
+        // read header
+        let header = BoxHeaderLite::read(reader)
+            .map_err(|_err| Error::InvalidAsset("Bad BMFF".to_string()))?;
+        if header.name != BoxType::StszBox {
+            return Err(Error::InvalidAsset("Bad BMFF".to_string()));
+        }
+
+        // read extended header
+        let (_version, _flags) = read_box_header_ext(reader)?; // box extensions
+
+        // get count
+        let entry_count = reader.read_u32::<BigEndian>()?;
+
+        // read and patch offsets
+        let mut entries: Vec<ChunkBox> = Vec::new();
+        for _e in 0..entry_count {
+            let first_chunk = reader.read_u32::<BigEndian>()?;
+            let samples_per_chunk = reader.read_u32::<BigEndian>()?;
+            let sample_description_index = reader.read_u32::<BigEndian>()?;
+
+            entries.push(ChunkBox {
+                first_chunk,
+                samples_per_chunk,
+                sample_description_index,
+            });
+        }
+
+        Ok(SampleToChunkBox {
+            entry_count,
+            entries,
+        })
+    }
+}
+
+#[allow(dead_code)]
```

Comment:
Is there a feature we should be using here?

Reply:
Dead code removed